lib/controllers/project.js

```diff
@@ -216,10 +216,51 @@ Project.prototype.saveVars = function(projectId, vars, callback) {
 
 //-----------------------------------------------------------------------------
 Project.prototype.deploy = function(projectId, dir, includeModules, registry, withTests, callback) {
-  var projectController = this;
-  var projectType;
+  var projectController = this, projectType;
+  var errMessage = 'You are deploying a %s project to a %s runtime. Please select the correct runtime.';
+
+  function parseVersion (str) {
+    if (typeof str != 'string') return false;
+    str = str.split('.');
+
+    return {
+      major: parseInt(str[0], 10) || 0,
+      minor: parseInt(str[1], 10) || 0,
+      patch: parseInt(str[2], 10) || 0
+    };
+  }
 
   async.waterfall([
+      projectController.detectProjectType.bind(null, dir),
+      function (type, cb) {
+        librarian.project.find({ projectId: projectId }, userConfig.data.apiKey, function (err, project) {
+          if (err) return callback(Errors.getMessage(err));
+          projectType = project.images.run.label;
+          if (projectType !== type) {
+            errMessage = util.format(errMessage, type, projectType);
+            return callback(errMessage);
+          }
+          cb();
+        });
+      },
+      function (cb) {
+        var versionFile, version;
+        var minMeteorVersion = parseVersion(Project.supportedVersions.Meteor);
+
+        if (projectType === 'Meteor') {
+          versionFile = path.join(dir, '.meteor', 'release');
+          fs.readFile(versionFile, { encoding: 'utf8' }, function (err, deployVersion) {
+            deployVersion = deployVersion.replace('METEOR@', '');
+            version = parseVersion(deployVersion);
+
+            if (version.major < minMeteorVersion.major ||
+                version.major >= minMeteorVersion.major && version.minor < minMeteorVersion.minor) {
```

Comment:
wrap conditional to make clear (or better: move to a function `isSupported(srcVersion, type)`